manifests/haproxy/autoload.pp

```diff
@@ -36,13 +36,9 @@
             $parts = [$options['_domain_prefix'], $env, $options['_domain_suffix']].filter |$item| { !$item.empty }
             $domain_env = join($parts, '.')
           }
-          file { [$path_env,
-            "$path_env/current",
-            "$path_env/current/redirects"]:
-            ensure => directory,
-            owner => $http_edge_path_owner,
-            replace => false
-          }
+          ensure_resource('file', $path_env, {'ensure' => 'directory', 'owner' => $http_edge_path_owner, 'replace' => 'false' })
+          ensure_resource('file', "$path_env/current", {'ensure' => 'directory', 'owner' => $http_edge_path_owner, 'replace' => 'false' })
```

Comment:
not a problem.

1. puppet creates the directory, with replace=false  and owner=deploy
2. ansistrano deploys the project. Removes current/ (because it is owned by deploy user) and creates a symlink
3. puppet won't touch current/, because of replace=false

Reply:
@NITEMAN Just tried out what would happen if `/current` folder was generated by the puppet code before launching the http-edge deploy job:

```
sbitio-kris@XXXX:~$ ls -lah /opt/http-edge/dev/
total 12K
drwxr-sr-x 3 deploy deploy 4,0K nov  3 18:13 .
drwxr-sr-x 8 deploy deploy 4,0K nov  3 18:13 ..
drwxr-sr-x 3 deploy deploy 4,0K nov  3 18:13 current
sbitio-kris@XXXX:~$ ls -lah /opt/http-edge/dev/
total 16K
drwxr-sr-x 4 deploy deploy 4,0K nov  3 18:17 .
drwxr-sr-x 8 deploy deploy 4,0K nov  3 18:13 ..
lrwxrwxrwx 1 deploy deploy   26 nov  3 18:17 current -> ./releases/20211103171749Z
drwxr-sr-x 3 deploy deploy 4,0K nov  3 18:17 releases
drwxr-sr-x 3 deploy deploy 4,0K nov  3 18:17 shared
```

These commands were launched 4 minutes apart. The first `ls` was done after executing puppet and before launching the Jenkins job, while the second one was done after launching the Jenkins job. We can see that there were no issues and the symlink was created successfully.